borg/helpers.py

```diff
@@ -1080,3 +1082,111 @@ def log_multi(*msgs, level=logging.INFO):
         lines.extend(msg.splitlines())
     for line in lines:
         logger.log(level, line)
+
+
+class ItemFormatter:
+    FIXED_KEYS = {
+        # Formatting aids
+        'LF': '\n',
+        'SPACE': ' ',
+        'TAB': '\t',
+        'CR': '\r',
+        'NUL': '\0',
+        'NEWLINE': os.linesep,
+    }
+
+    @classmethod
+    def available_keys(cls):
+        class FakeArchive:
+            name = ""
+
+        formatter = cls(FakeArchive, "")
+        keys = []
+        keys.extend(formatter.call_keys.keys())
+        keys.extend(formatter.get_item_data({
+            b'mode': 0, b'path': '', b'user': '', b'group': '', b'mtime': 0,
+            b'uid': 0, b'gid': 0,
+        }).keys())
```

Comment:
oh, I was still thinking in lists, like in python2. yes in py3, that does not work.

that would work, not sure if it helps with readability:

fake_item = {...}
keys = set(formatter.call_keys) | set(formatter.get_item_data(fake_item))